### PR TITLE
[711] Smoke test for /healthcheck and /courses routes

### PR DESCRIPTION
**Pull request by AbigailMcP**
### Context

https://trello.com/c/tcSBRSoL/711-m-healthcheck-smoke-test
https://trello.com/c/2EUGHq7k/713-m-courses-endpoint-smoke-tests

### Changes proposed in this pull request

This PR is ⭐inspired⭐  by https://github.com/DFE-Digital/register-trainee-teachers/pull/372:

- Create a new spec directory for smoke tests
- Setup basic healthcheck and courses smoke tests to assert response of system without stubs
- Setup a tag to run specs without smoke tests by default
- Run smoke tests in build workflow

### Checklist

- [x] Make sure all information from the Trello card is in here
- [x] Attach to Trello card
- [x] Rebased master
- [x] Cleaned commit history
- [x] Tested by running locally

### CODE REVIEW

**Comment by AbigailMcP** [comment]
[[M] Healthcheck smoke test](https://trello.com/c/tcSBRSoL/711-m-healthcheck-smoke-test)

File: spec/requests/smoke/health_check_spec.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+describe "system health check spec", :smoke do
+  describe "GET /healthcheck" do
```
Comment (misaka) [inline spec/requests/smoke/health_check_spec.rb:6]: Is there a difference between the tests here and the existing tests?

https://github.com/DFE-Digital/teacher-training-api/blob/711-healthcheck-smoke-test/spec/requests/heartbeat_spec.rb#L12

Comment (darokel) [reply spec/requests/smoke/health_check_spec.rb:6]: Yes, this one does not stub out any data whereas the other tests different scenarios based on different errors which **could** occur

**Comment by AbigailMcP** [comment]
@darokel - I wasn't sure where to look to grab all the base urls for the different environments - but these need to be added to the setting files.

**Comment by AbigailMcP** [comment]
[[M] Courses endpoint smoke tests](https://trello.com/c/2EUGHq7k/713-m-courses-endpoint-smoke-tests)

File: spec/requests/smoke/health_check_spec.rb (hunk 1 above)
Comment (misaka) [reply spec/requests/smoke/health_check_spec.rb]: Aaahh. *sound-of-penny-dropping* ... this is only run when the smoke tests run. Makes sense now. 👍

**Comment by AbigailMcP** [comment]
Closing in favour of https://github.com/DFE-Digital/teacher-training-api/pull/1728

**Pull request closed without merging**